### DISCUSSION TITLE
cannot open source file "MooseApp.h"

### DISCUSSION THREAD

**Discussion by nan-f** [idaholab/moose#25684] [Q&A Getting Started]
Hi all,
I already had MOOSE on my system,and I try to write code on vscode. but I can’t find source file "MooseApp.h".
This is my setting.json:
```
{
    "[cpp]": {
        "editor.defaultFormatter": "ms-vscode.cpptools"
    },
    "telemetry.telemetryLevel": "off",
    "C_Cpp.default.cppStandard": "c++17",
    "C_Cpp.default.includePath": [
        "${workspaceFolder}/**",
        "/home/fangn/mambaforge3/envs/moose/**",
        "/usr/include/**"
    ],
    "files.associations": {
        "*.C": "cpp",
        "*.h": "cpp"
    },
    "files.trimTrailingWhitespace": true,
    "editor.tabSize": 2,
    "editor.formatOnSave": true,
    "files.insertFinalNewline": true
}
```

**Comment by GiudGiud** [comment]
Hello

I just tried with your settings and it works fine. Ctrl + P will still find MooseApp.h

Are you getting a compiling error? This would not be related to your vscode settings

Guillaume

**Reply by nan-f** [reply]
I try to change my  `.vscode/c_cpp_properties.json` as  https://github.com/idaholab/moose/discussions/18307 say, and then I can find MooseApp.h. :smile:
